### ISSUE TITLE
UnidentifiedImageError for some items in https://huggingface.co/datasets/BLIP3o/BLIP3o-Pretrain-JourneyDB

### ISSUE THREAD

**Issue by alfassy** [JiuhaiChen/BLIP3o#65]
Hi dear authors,
Thank you for the great paper, I think it is a breakthrough to be able to add such image generation abilities without hurting the image understanding abilities of the model.
I am trying to train BLIP3O from scratch using Qwen2.5VL and eva-clip as explained in the README.
I downloaded the data using the suggest "python -c "from huggingface_hub import snapshot_download; print(snapshot_download(repo_id='BLIP3o/BLIP3o-Pretrain', repo_type='dataset'))" " 
Then tried to train in webdataset format. 
pointing the suggested glob.glob to all tar object under my HF folder /amit/HF/hub/datasets--BLIP3o--BLIP3o-Pretrain-JourneyDB/snapshots/734b390be883b2021ae6d6dc1b24e6b97e7253a2/*.tar and all are picked up. I train with "--data_type mix"
The model in learning for a few batches and loss goes down until I met this error:
'cannot identify image file <_io.BytesIO object at 0x147775b51cb0>' (Error Type: UnidentifiedImageError)
I tried to redownload the data to no avail, for now I have a try catch but I if I am doing something wrong then I prefer to fix it.
My final goal is to add another VLM as an option and publish to open source so I'd appreciate the help. (For now I am only asking about the basic Qwen and eva-clip to form a working baseline)
Have a nice week and thank you in advance,
Amit

**Comment by JiuhaiChen** [comment]
Hi, thanks for your interest. It looks weird, can you repeat this error? And we use try except in the dataloader:

https://github.com/JiuhaiChen/BLIP3o/blob/main/blip3o/train/train.py#L691

The open error should can be skipped.

**Comment by alfassy** [comment]
Hi,
Thank you for your answer.
So I actually get the error at the start of the __getitem__() and that's where I had to add the try catch.
So I understand that's normal that some images can't be opened and are skipped, if that's the case then you can close the issue. 
Thank you and have a nice weekend,
Amit

<img width="702" height="233" alt="Image" src="https://github.com/user-attachments/assets/55191181-09b1-4c33-b357-51535255324a" />

**Comment by alfassy** [comment]
<img width="1159" height="240" alt="Image" src="https://github.com/user-attachments/assets/70d9bb0a-11cf-440d-9eb4-9dd9d2b808c5" />
My solution for now

**Comment by alfassy** [comment]
I just stayed with the same solution, skipping those samples.
